src/nunavut/lang/c/support/nunavut.j2

```diff
@@ -0,0 +1,504 @@
+/*
+ *
+ * UAVCAN common serialization support routines.                                                                +-+ +-+
+ *
+ * Not meant to be included directly by a user. Please include nunavut.h
+ * or nunavut_le.h instead, depending on your platform architecture.
+ *
+ *                                                                                                            \  -  /
+ *  This software is distributed under the terms of the MIT License.                                            ---
+ *                                                                                                               o
+ * +------------------------------------------------------------------------------------------------------------------+
+ */
+#ifndef NUNAVUT_SUPPORT_NUNAVUT_COMMON_H_INCLUDED
+#define NUNAVUT_SUPPORT_NUNAVUT_COMMON_H_INCLUDED
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <string.h>
+#include <float.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef uint32_t NunavutUnsignedBitLength;
+typedef int32_t  NunavutSignedBitLength;
+typedef float    NunavutFloat32;
+typedef double   NunavutFloat64;
+
+// appropriate name?
+// do we typedef this, leave it anon, or enum NAME
+#define NUNAVUT_ERR_INVALID_BUF 1
+#define NUNAVUT_ERR_INVALID_LEN 2
+#define NUNAVUT_ERR_INVALID_OFFSET 3
+#define NUNAVUT_ERR_INVALID_TAG 4
+
+/// By default, this macro resolves to the standard assert(). The user can redefine this if necessary.
+/// To disable assertion checks completely, make it expand into `(void)(0)`.
+#ifndef NUNAVUT_ASSERT
+// Intentional violation of MISRA: assertion macro cannot be replaced with a function definition.
+#    define NUNAVUT_ASSERT(x) assert(x)  // NOSONAR
+#endif
+
+#if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
```

Comment:
This check fails when compiled with g++, a solution is to just add a check for `__cplusplus`, but I'm not sure what exactly this should support.

Reply:
This is actually a check for the version of the C compiler. I think we should not build them using g++ as this PR is for C serialization support and not C++ (I think that's on another PR?)

### Additional info:

As mentioned [here](https://gcc.gnu.org/onlinedocs/cpp/Standard-Predefined-Macros.html), __STDC_VERSION__ is not defined when building on g++.

This check came from the fact that libcanard was released to support only C11 or later. I actually issued a [PR on libcanard](https://github.com/UAVCAN/libcanard/pull/151) relaxing the support of the library to C99.

Assuming @pavel-kirienko and @thirtytwobits allows Nunavut-generated C-codes to be backwards-compatible with C99, we might have to apply the same changes I made to `canard_dsdl.h` to your `nunavut.j2`.

Reply:
This check is to ensure that the static assertion checks are available. It can be replaced with dedicated checks for C++ and C to fix the issue.

> I think we should not build them using g++ 

This is incorrect because C headers are commonly included into C++ translation units, so we have to support both.

Reply:
I see. So the header will be used for both C and C++.

Reply:
I'm still not sure what the solution should be. Are we supporting a certain version of C++? For now I've just added an extra layer to avoid this check if `__cplusplus` is defined.

Reply:
FYI: in my sandbox branch:

https://github.com/thirtytwobits/nunavut/tree/issue/115

I've added support for building pure C tests in addition to the c++-using-c-tests.

Reply:
perhaps:
```
#ifdef __cplusplus
#   if (__cplusplus < 201100L)
#       error "Unsupported language: ISO C11, C++11, or a newer version of either is required."
#   endif
extern "C"
{
#else
#   if !defined(__STDC_VERSION__) || (__STDC_VERSION__ < 201112L)
#       error "Unsupported language: ISO C11 or a newer version is required."
#   endif
#endif
```

Reply:
Also use `static_assert` instead of `_Static_assert` since this is a convenience method in assert.h and also compatible with C++.